front-end/src/components/UploadForm.tsx

```diff
@@ -17,73 +18,85 @@ function UploadForm(){
     event.preventDefault();
     if(validToken()){
       //send token to backend
+      //if successful, display success message
+      setSubmitStatus(true);
     }
     else {
       //display error pop-up message
     }
   }
 
+  function handlePopUpClick(){
+    setSubmitStatus(false);
+  }
   return (
     <div>
-      {/**Header section*/}
-      <section className="section is-medium has-text-centered">
-        <div className="container">
-          <h1 className="title">Sharing my close contacts with health care professional</h1>
-          <h2 className="subtitle">
-            Please enter the health care code provided to you by the health care professional
-          </h2>
-          
-          {/*Upload Form*/}
-          <form onSubmit={handleSubmit}>
-            {/*Input column*/}
-            <div className="columns is-centered">
-              <div className="column is-half">
-                <input 
-                  className="input" 
-                  type="text" 
-                  placeholder="Please insert the token here"
-                  onChange={e => setVerificationToken(e.target.value) }
-                />
-                {!validToken() && verificationToken && <p className="help is-danger">
-                  Please insert a valid token
-                </p>}
-              </div>
-            </div>
-      
-            {/*Text below the text input*/}
-            <h6 className="subtitle is-6">
-              Only authorised health care professionals can provide a valid health care code
-            </h6>
       
-            <div className="columns is-centered">
-              {/*Cancel button*/}
-              <div className="column is-half">
-                <div className="field">
-                  <p className="control">
-                    <button className="button">
-                      Cancel
-                    </button>
-                  </p>
+      {/**Notification that appears when the token submission is successful*/}
```

Comment:
It looks like there are two different pages with no shared content? In that case, you can just have a conditional return statement which makes this really clear:

```
if (submitSuccess) {
return (
 <div className="column is-half"> <UploadSuccess onPopUpClick={handlePopUpClick}/>...
)
} 

else {
return (
<section className="section is-medium has-text-centered">...
}

```

Your solution achieves the same thing, this is just a little clearer.

Reply:
Thanks for the tip Jake!
I will keep that in mind 👍